python/ray/tune/analysis/experiment_analysis.py

```diff
@@ -122,12 +124,20 @@ def _load_trials(self) -> List[Trial]:
         with self._fs.open_input_stream(self._experiment_json_fs_path) as f:
             experiment_state = json.loads(f.readall(), cls=TuneFunctionDecoder)
 
+        experiment_fs_path = Path(self._experiment_fs_path)
+
         trials = []
         trial_states = experiment_state["trial_data"]
         for trial_json_state, trial_runtime_metadata in trial_states:
             trial = Trial.from_json_state(trial_json_state, stub=True)
             trial.restore_run_metadata(trial_runtime_metadata)
-            # TODO(justinvyu): [handle_moved_storage_path]
+
+            new_storage = copy.copy(trial.storage)
```

Comment:
What would happen if a experiment directory originally on S3, then moved to disk. Will we automatically detect the and build a new file system?

Reply:
`Tuner.restore("local_path")` would pass in a local storage path and would be resolved into a local filesystem. I should add a test for this.